aws-api/src/main/java/com/amplifyframework/api/aws/MutiAuthSubscriptionOperation.java

```diff
@@ -116,7 +116,20 @@ private void dispatchRequest() {
                     MutiAuthSubscriptionOperation.this.subscriptionId = subscriptionId;
                     onSubscriptionStart.accept(subscriptionId);
                 },
-                onNextItem,
+                response -> {
+                    if (response.hasErrors()) {
+                        // If there are auth-related errors, dispatch an ApiAuthException
+                        if (hasAuthRelatedErrors(response)) {
+                            executorService.submit(this::dispatchRequest);
+                            return;
+                        }
+                        // Otherwise, we just want to dispatch it as a next item and
+                        // let callers deal with the errors.
+                        onNextItem.accept(response);
+                    } else {
+                        onNextItem.accept(response);
+                    }
+                },
```

Comment:
Now that the error handling logic is in the response callback, can the similar logic (`executorService.submit(this::dispatchRequest);`) be removed from the error callback?

Reply:
`ApiAuthException` can still be thrown when `SubscriptionEndpoint` calls `createHeadersForSubscription` in the authorizer. Specifically if they're trying to use OIDC without having passed in a provider at configuration time.